internal/cluster/validations/validations.go

```diff
@@ -312,73 +315,349 @@ func ValidateVipDHCPAllocationWithIPv6(vipDhcpAllocation bool, machineNetworkCID
 	return nil
 }
 
-func ValidateIPAddresses(ipV6Supported bool, obj interface{}) error {
-	var allAddresses []*string
-	var apiVip string
-	var ingressVip string
-	var machineNetworks []*models.MachineNetwork
-	userManagedNetworking := false
-	vipDhcpAllocation := false
+func handleApiVipBackwardsCompatibility(clusterId strfmt.UUID, apiVip string, apiVips []*models.APIVip) ([]*models.APIVip, error) {
+	// APIVip provided, but APIVips were not.
+	if apiVip != "" && len(apiVips) == 0 {
+		return []*models.APIVip{{IP: models.IP(apiVip), ClusterID: clusterId}}, nil
+	}
+	// Both APIVip and APIVips were provided.
+	if apiVip != "" && len(apiVips) > 0 && apiVip != string(apiVips[0].IP) {
+		return nil, errors.New("apiVIP must be the same as the first element of apiVIPs")
+	}
+	// APIVips were provided, but APIVip was not.
+	if apiVip == "" && apiVips != nil && len(apiVips) > 0 {
+		return nil, errors.New("request must include apiVIP alongside apiVIPs")
```

Comment:
Why the new client actually should send both singular and plural variant? Isn't it enough for a new client to send only the plural one? I.e. don't we want to deprecate the singular variant?

Reply:
It is best explained in @mkowalski's doc: https://github.com/openshift/assisted-service/pull/4245/files#diff-0f2f3122fee0aeba312f537e745da06d17e1c4588f7ea6177834bbf2bfa161ceR74

Reply:
Yeah seen that, just didn't understand why getting a plural field isn't a good indication for a new client:/ I'll try to go over the doc again...